### PR TITLE
Update angular-resource.d.ts

### PR DESCRIPTION
**Pull request by khoden**
fix IResourceArray. It is array of IResource<T>, not just array of T

### CODE REVIEW

File: angularjs/angular-resource.d.ts [hunk 1]
```diff
@@ -141,7 +141,7 @@ declare module angular.resource {
     /**
      * Really just a regular Array object with $promise and $resolve attached to it
      */
-    interface IResourceArray<T> extends Array<T> {
+    interface IResourceArray<T> extends Array<IResource<T>> {
```
Comment (vvakame) [inline angularjs/angular-resource.d.ts]: I think `Array<T & IResource<T>>` is more better. what do you think?

Comment (KhodeN) [inline angularjs/angular-resource.d.ts]: Agree

**Comment by vvakame** [comment]
thanks mate!

**Comment by mayankn** [comment]
@vvakame @KhodeN This latest merge is creating some errors. Can someone look at these urgently?:
[TypeScript] TypeScript error: /.tmp/typings/angularjs/angular-resource.d.ts(147,49): error TS1005: ',' expected.
[TypeScript] TypeScript error: /.tmp/typings/angularjs/angular-resource.d.ts(149,55): error TS1005: '(' expected.
[TypeScript] TypeScript error: /.tmp/typings/angularjs/angular-resource.d.ts(150,28): error TS1005: ',' expected.
TypeScript: 3 syntax errors
TypeScript: emit failed

**Comment by KhodeN** [comment]
@mayankn What is your tsc version?
Tests with Travis CI are passed succesfully, there is no syntax errors.

**Comment by mayankn** [comment]
$ tsc
Version 1.6.2

**Comment by mayankn** [comment]
Sorry, actually my project uses gulp-typescript 2.8.1, which i believe is tsc 1.5.3

**Comment by mayankn** [comment]
updated gulp-typescript to latest v2.10.0, which uses tsc 1.7.3 to fix the issue. No more errors. But I would've thought this commit should've been backward compatible with older tsc versions as well

**Comment by KhodeN** [comment]
old typescript doesn't support [Intersection Types](https://github.com/Microsoft/TypeScript/blob/master/doc/spec.md#3.5) feature.

It is very usefull for that kind of api as $resource entity

**Pull request merged**